rubin_changelog/changelog.py

```diff
@@ -0,0 +1,182 @@
+        result = SortedDict()
+        releases = eups_data['releases']
+        last_release = None
+        for r in releases:
+            if last_release is not None:
+                previous_pkgs = [sub['package'] for sub in last_release]
+                pkgs = [sub['package'] for sub in releases[r]]
+                removed = SortedList(set(previous_pkgs) - set(pkgs))
+                added = SortedList(set(pkgs) - set(previous_pkgs))
+                result[r] = {'added': added, 'removed': removed}
+            last_release = releases[r]
+        return result
+
+    @staticmethod
+    def _fetch(repo: str) -> dict:
+        log.info("Fetching %s", repo)
+        gh = GitHubData()
+        result = dict()
+        pulls = gh.get_pull_requests(repo)
+        tags = gh.get_tags(repo)
+        result["repo"] = repo
+        result["pulls"] = pulls
+        result["tags"] = tags
+        del gh
+        return result
+
+    def _get_package_repos(self, products: SortedList) -> SortedDict:
+        result = SortedDict()
+        result['pulls'] = SortedDict()
+        result["tags"] = SortedDict()
+        gh = GitHubData()
+        repos = gh.get_repos()
+        del gh
+        repo_list = SortedList()
+        for product in products:
+            if product in repos:
+                repo_list.add(product)
+        with ThreadPoolExecutor(
+                max_workers=self._max_workers) as executor:
+            futures = {executor.submit(self._fetch, repo): repo for repo in repo_list}
+            for future in concurrent.futures.as_completed(futures):
+                try:
+                    data = future.result()
+                except Exception:
+                    log.error("Fetch failed")
+                else:
+                    repo = data["repo"]
+                    result["pulls"][repo] = data["pulls"]
+                    result["tags"][repo] = data["tags"]
+        return result
+
+    def get_package_repos(self, products: SortedList, release: ReleaseType) -> SortedDict:
+        if self._github_cache is None:
+            self._github_cache = self._get_package_repos(products)
+        else:
+            log.info("Using cached github data")
+        cache = self._github_cache
+        result = SortedDict()
+        result["tags"] = SortedDict()
+        result['pulls'] = cache['pulls']
+        repo_result = SortedDict()
+        for repo in cache['tags']:
+            repo_result[repo] = list()
+            for tag in cache['tags'][repo]:
+                rtag = Tag(tag["name"])
+                if rtag.is_valid() and matches_release(rtag, release):
+                    repo_result[repo].append(tag)
+        result["tags"] = repo_result
+        return result
+
+    @staticmethod
+    def _ticket_number(title: str) -> int:
+        match = re.search(r'DM[\s*|-]\d+', title.upper())
+        ticket = None
+        if match:
+            res = re.findall(r"DM[\s|-]*(\d+)", match[0])
```

Comment:
If you put the capture group in the first regexp, couldn't this be as simple as `return match[1]`?